app/controllers/webauthn_verifications_controller.rb

```diff
@@ -35,22 +35,25 @@ def authenticate
 
     redirect_to(URI.parse("http://localhost:#{port}?code=#{@verification.otp}").to_s, allow_other_host: true)
   rescue WebAuthn::Error => e
-    # TODO: render plain text for both endpoints
-    render json: { message: e.message }, status: :unauthorized
+    render plain: e.message, status: :unauthorized
   rescue ActionController::ParameterMissing
-    render json: { message: "Credentials required" }, status: :unauthorized
+    render plain: t("credentials_required"), status: :unauthorized
   ensure
     session.delete(:webauthn_authentication)
   end
 
+  def failed_verification
+    @message = params.permit(:error).fetch(:error, "")
+    Rails.logger.info("[webauthn_verification:authenticate] Verification failed: #{@message}")
```

Comment:
Displaying and logging message as [suggested](https://bundler.slack.com/archives/C038T4BU73M/p1680276231913739): 

> can we add a log or something in the catch, rather than completely discarding the error?

> Are you thinking of logging it our side (Rails logs) or like displaying a message to the user on why it failed (or both).

> Hmmm I think both would be good? We could pass the error in the query param when redirecting to the failure page?